### ISSUE TITLE
mongoose populate pass found object reference key

### ISSUE THREAD

**Issue by shivshankar3578** [Automattic/mongoose#5817]
I have a mongoose Group schema which contains invitee (array of sub document) and currentMove, invitee also contains currentMove and I want to get document with only sub document that have same currentMove.

```
Group.findById("5a03fa29fafa645c8a399353")
.populate({
    path: 'invitee.user_id',
    select: 'currentMove',
    model:"User",
     match: {
         "currentMove":{
            $eq: "$currentMove"
        }
    }
}) 
```

This generates unknown currentMove Object id for match query. I'm not sure if mongoose has this functionality. Can anyone help me, please?

**Comment by vkarpov15** [comment]
No mongoose doesn't have support for referencing fields in `$match`. The right hand side of `$eq` must be something you know ahead of time, but definitely a feature to consider adding for future releases.

**Comment by vkarpov15** [comment]
I believe this is fixed in Mongoose 5.5 with [populate match functions](https://thecodebarbarian.com/mongoose-5-5-static-hooks-and-populate-match-functions#populate-match-function).

```javascript
Group.findById("5a03fa29fafa645c8a399353")
.populate({
    path: 'invitee.user_id',
    select: 'currentMove',
    model:"User",
     match: doc => ({ currentMove: doc.currentMove })
}) 
```
